component/main.jsonnet

```diff
@@ -5,6 +5,61 @@ local inv = kap.inventory();
 // The hiera parameters for the component
 local params = inv.parameters.machineset_egress_cidr_operator;
 
+local namespace = kube.Namespace(params.namespace);
+
+local serviceAccount = kube.ServiceAccount('machineset-egress-cidr-operator') {
+  metadata+: {
+    namespace: params.namespace,
+  },
+};
+
+local clusterRole = kube.ClusterRole('machineset-egress-cidr-operator') {
+  rules: params.clusterRole_rules,
+};
+
+local clusterRoleBinding = kube.ClusterRoleBinding('machineset-egress-cidr-operators') {
+  subjects_: [serviceAccount],
+  roleRef_: clusterRole,
+};
+
+local roleBindnig = kube.RoleBinding('machineset-egress-cidr-operators') {
+  metadata+: {
+    namespace: params.namespace,
+  },
+  subjects_: [serviceAccount],
+  roleRef: {
+    kind: 'Role',
+    name: 'leader-election-role',
+  },
+};
+
+local deployment = kube.Deployment('machineset-egress-cidr-operators') {
+  metadata+: {
+    namespace: params.namespace,
+    labels: {
+      'app.kubernetes.io/name': 'machineset-egress-cidr-operator',
+      'app.kubernetes.io/managed-by': 'syn',
```

Comment:
I recommend to define the common labels in the parameters and then you can reuse it for every object like `lables: params.labels,`

You also might want to add more labels (see https://syn.tools/syn/references/labels-guide.html)